core/src/main/java/org/mvcspec/ozark/core/ViewResponseFilter.java

```diff
@@ -173,6 +179,24 @@ public void filter(ContainerRequestContext requestContext,
         }
     }
 
+    /*
+     * Append to view name default extension if one available and applicable.
+     */
+    protected static String appendExtensionIfRequired(String viewName, String defaultExtension) {
+        if (viewName == null || viewName.startsWith(REDIRECT)
+            || defaultExtension == null || defaultExtension.length() == 0) {
+            return viewName;
+        }
+
+        String resultView = viewName;
+        if (!viewName.contains(DOT_CHAR) && !viewName.contains(":")) {
```

Comment:
Could you explain the `!viewName.contains(":")` here? Not sure if this is really needed. Especially because I know at least one view engine which supports view names like `react:some-folder/foobar.jsp`.

Reply:
Exactly for view engines like mentioned I skipped the default extension attaching in case if the character `:` available in the incoming view name. So, view names like `react:some-folder/foobar.jsp` or `react:some-folder/foobar` will be excluded from handling and remain unchanged.
Do you think that it is an incorrect assumption and we still need to process such view names?

Reply:
I don't see a reason why we should not apply the default file extension mechanism if the view name contains `:`. Actually a view name like `react:foobar.jsp` would mean something like "use the react view engine but render the view using a default JSP view engine". So why not apply the default mechanism in this case?

Of course `redirect:` needs special treatment, but this is already done in the first `if` block. So I think it is safe to add the default extension if there is no `.` char in the view name. So I would prefer to simply remove the `!viewName.contains(":")` part.

Reply:
OK, I got it. I made the required changes.